packages/core/content-type-builder/server/src/controllers/validation/types.ts

```diff
@@ -73,11 +82,7 @@ const getTypeShape = (attribute, { modelType, attributes } = {}) => {
             'cannot define a default UID if the targetField is set',
             function (value) {
               const { targetField } = this.parent;
-              if (_.isNil(targetField) || _.isNil(value)) {
-                return true;
-              }
-
-              return false;
+              return !!(_.isNil(targetField) || _.isNil(value));
```

Comment:
```suggestion
              return _.isNil(targetField) || _.isNil(value);
```
isNil only returns booleans